src/app/Command.cpp

```diff
@@ -40,8 +40,8 @@ CHIP_ERROR Command::Init(Messaging::ExchangeManager * apExchangeMgr)
     VerifyOrExit(mpExchangeCtx == nullptr, err = CHIP_ERROR_INCORRECT_STATE);
 
     mpExchangeMgr = apExchangeMgr;
-
-    err = Reset();
+    mpDelegate    = apDelegate;
+    err           = Reset();
```

Comment:
nit for the future: in other pieces of the code, Reset() is the inverse of Init(), which would generally lead me to believe that delegate/manager would be cleared. 

The existing Command.h Reset does not even have documentation, but it does move the state to Initialize().

Maybe it should be called initialize after all ... or ReInitialize() or something more intuitive.

Reply:
sure, thanks for suggestion, create ticket for this.
https://github.com/project-chip/connectedhomeip/issues/5623